pkg/update/update_test.go

```diff
@@ -198,3 +198,39 @@ func TestParseUpdateType(t *testing.T) {
 		})
 	}
 }
+
+func TestCompareVersions(t *testing.T) {
+	tests := []struct {
+		name string
+		a    []string
+		b    []string
+		want bool
+	}{
+		{
+			name: "Test Case 1 - Equal slices",
+			a:    []string{"v1.0.0", "v1.1.0", "v1.2.0"},
```

Comment:
We aren't comparing just versions. We are comparing packages so it should be `pkg1@~v1.0.0` , `pkg2@~v1.4.0` , etc.

Reply:
Speaking of which, the function name should also be `ComparePackages`